docs/adr/adr-009-public-api.md

```diff
@@ -0,0 +1,214 @@
+* Disassociate centralization. Services have always been associated with centralized infrastructure with older 
+monolithic services and newer distributed microservice architectures. 
+
+#### Modules Definition
+
+##### Header
+```go
+type HeaderModule interface {
+	// Head returns the node's local head (tip of the chain of the header store).
+    Head(ctx context.Context) (*header.ExtendedHeader, error)
+	// Get returns the header of the given hash from the node's header store.
+    Get(ctx context.Context, hash tmbytes.HexBytes) (*header.ExtendedHeader, error)
+	// GetByHeight returns the header of the given height from the node's header store.
+	// If the header of the given height is not yet available, the request will hang
+	// until it becomes available in the node's header store.
+    GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error)
+    // GetRangeByHeight returns the given range [from:to) of ExtendedHeaders.
+    GetRangeByHeight(ctx context.Context, from, to uint64) ([]*ExtendedHeader, error)
+    // Subscribe creates long-living Subscription for validated ExtendedHeaders.
+    // Multiple Subscriptions can be created.
+    Subscribe() (Subscription, error)    
+	// SyncState returns the current state of the header Syncer. 
+	SyncState() sync.State
+	// SyncWait blocks until the header Syncer is synced to network head. 
+    SyncWait(ctx context.Context) error
+	// SyncHead provides the Syncer's view of the current network head.
+	SyncHead(ctx context.Context) (*header.ExtendedHeader, error)
+}
+```
+
+##### Shares
+```go
+type SharesModule interface {
+	// GetShare returns the Share from the given data Root at the given row/col coordinates.
+	GetShare(ctx context.Context, dah *Root, row, col int) (Share, error)
+	// GetSharesByNamespace returns all shares of the given nID from the given data Root.
+    GetSharesByNamespace(ctx context.Context, root *Root, nID namespace.ID) ([]Share, error)
+    // SharesAvailable subjectively validates if Shares committed to the given data Root are 
+	// available on the network. 
+    SharesAvailable(ctx context.Context, root *Root) error
+    // ProbabilityOfAvailability calculates the probability of the data square
+    // being available based on the number of samples collected.
+    ProbabilityOfAvailability() float64
+}
+```
+
+##### P2P
+```go
+type P2PModule interface {
+    Info
+    Peers
+    PeerInfo
+
+    Connect
+    Disconnect
+    Connectedness
+    NATStatus
+
+    BlockAdd
+    BlockRm
+    BlockList
+
+    MutualAdd
+    MutualRm
+    MutualList
+
+    BandwidthStats
+    BandwidthPeer
+    BandwidthProtocol
+
+    ResourceState
+    ResourceLimit
+    ResourceSetLimit
+
+    PubSubPeers
+    PubSubScore
+
+    BitSwapStats
+    BitSwapLedger
+
+    DHTFindPeer
+}
+```
+
+##### Wallet
+```go
+type WalletModule interface {
+    New
+    Del
+    Has
+    List
+    Sign
+    Export
+    Import
+    // TODO: Mnemonics stuff?
+}
+```
```

Comment:
I want to move the wallet module to _nice to have_ because we have the `cel-key` utility which is good ~enough~ for now.

Reply:
This separation is good.

Also, "key utility" is IMHO way better name than "WalletModule" (because wallets typically shows your bags, let you see history, do transfers, etc).